huaweicloud/resource_huaweicloud_rds_read_replica_instance.go

```diff
@@ -0,0 +1,644 @@
+package huaweicloud
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/huaweicloud/golangsdk"
+	"github.com/huaweicloud/golangsdk/openstack/common/tags"
+	"github.com/huaweicloud/golangsdk/openstack/rds/v1/datastores"
+	"github.com/huaweicloud/golangsdk/openstack/rds/v1/flavors"
+	instanceV1 "github.com/huaweicloud/golangsdk/openstack/rds/v1/instances"
+	"github.com/huaweicloud/golangsdk/openstack/rds/v3/instances"
+)
+
+func resourceRdsReadReplicaInstance() *schema.Resource {
+
+	return &schema.Resource{
+
+		Create: resourceRdsReadReplicaInstanceCreate,
+		Read:   resourceRdsReadReplicaInstanceRead,
+		Update: resourceRdsReadReplicaInstanceUpdate,
+		Delete: resourceRdsReadReplicaInstanceDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceReadReplicaInstanceImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"primary_instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"flavor": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
```

Comment:
tips: `ForceNew` only supports true, so we can ignore it if the argument can be updatable.

Reply:
Got it, I have removed it in this resource.